lib/agent0/agent0/hyperdrive/crash_report/crash_report.py

```diff
@@ -0,0 +1,114 @@
+    # pylint: disable=too-many-return-statements
+
+    def default(self, o):
+        r"""Override default behavior"""
+        if isinstance(o, set):
+            return list(o)
+        if isinstance(o, HexBytes):
+            return o.hex()
+        if isinstance(o, (AttributeDict, MutableAttributeDict)):
+            return dict(o)
+        if isinstance(o, np.integer):
+            return int(o)
+        if isinstance(o, np.floating):
+            return float(o)
+        if isinstance(o, np.ndarray):
+            return o.tolist()
+        if isinstance(o, FixedPoint):
+            return str(o)
+        if isinstance(o, NumpyGenerator):
+            return "NumpyGenerator"
+        if isinstance(o, datetime):
+            return str(o)
+        try:
+            return o.__dict__
+        except AttributeError:
+            pass
+        # Let the base class default method raise the TypeError
+        return json.JSONEncoder.default(self, o)
+
+
+def setup_hyperdrive_crash_report_logging(log_format_string: str | None = None) -> None:
+    """Create a new logging file handler with CRITICAL log level for hyperdrive crash reporting.
+
+    In the future, a custom log level could be used specific to crash reporting.
+
+    Arguments
+    ---------
+    log_format_string : str, optional
+        Logging format described in string format.
+    """
+    logs.add_file_handler(
+        logger=None,  # use the default root logger
+        log_filename="hyperdrive_crash_report.log",
+        log_format_string=log_format_string,
+        delete_previous_logs=False,
+        log_level=logging.CRITICAL,
+    )
+
+
+# TODO: don't use ape's TransactionError, use web3's InvalidTransaction when we switch
+def log_hyperdrive_crash_report(
+    trade_type: str,
+    error: Exception,
+    amount: int,
+    agent_address: str,
+    pool_info: dict[str, Any],
+    pool_config: dict[str, Any],
+):
+    # pylint: disable=too-many-arguments
+    """Log a crash report for a hyperdrive transaction.
+    Arguments
+    ---------
+    trade_type : str
+        The type of trade being executed.
+    error : TransactionError
+        The transaction error that occurred.
+    amount : float
+        The amount of the transaction.
+    agent_address : str
+        The address of the agent executing the transaction.
+    pool_info : PoolInfo
+        Information about the pool involved in the transaction.
+    pool_config : PoolConfig
+        Configuration of the pool involved in the transaction.
+    Returns
+    -------
+    None
+        This function does not return any value.
+    """
+
+    # We remove the 'fees' object since the tuple is already expanded from the api
+    pool_config = pool_config.copy()
+    pool_config.pop("fees")
+    formatted_pool_info = json.dumps(pool_info, indent=4, cls=ExtendedJSONEncoder)
+    formatted_pool_config = json.dumps(pool_config, indent=4, cls=ExtendedJSONEncoder)
+
+    # TODO set up logging in file handler
```

Comment:
I believe this TODO is resolved now.

Reply:
Not quite, it's not logging to a file (I actually don't think any of our stuff is logging to files currently), only logging to stdout. Hoping to fix this in future PR

Reply:
Let's chat -- logging to file is set up, I'll show you

Reply:
ok we talked you should delete this TODO and we can fix the log-to-file from examples